src/Orleans.Persistence.CosmosDB/CosmosDBGrainStorage.cs

```diff
@@ -153,10 +153,12 @@ public async Task ReadStateAsync(string grainType, GrainReference grainReference
                 if (doc.Resource.State != null)
                 {
                     grainState.State = JsonConvert.DeserializeObject(doc.Resource.State.ToString(), grainState.State.GetType(), this._options.JsonSerializerSettings);
+                    grainState.RecordExists = true;
                 }
                 else
                 {
                     grainState.State = Activator.CreateInstance(grainState.State.GetType());
+                    grainState.RecordExists = true;

```

Comment:
@ReubenBond I think this is supposed to be false, per this original PR: https://github.com/OrleansContrib/Orleans.CosmosDB/pull/48/files#diff-f6c48f97559f4c1c65df1b87a7db74cc98308c699c48d0b4a10da1ad0c463272R161

as well as the MockStorageProvider from the tests, which sets it to false if there was no prior state: https://github.com/dotnet/orleans/blob/154feab1d49ee5c8e2590832d40a32e5c9ad4b26/test/TestInfrastructure/TestExtensions/MockStorageProvider.cs#L207

Reply:
Ah, you're right. Thanks for catching that, @davidvanchu! Are you able to open a PR to fix it?